packages/terra-date-picker/src/react-datepicker/index.jsx

```diff
@@ -562,6 +565,12 @@ class DatePicker extends React.Component {
     }
   }
 
+  onInputKeyDown = (event) => {
+    if(event.key === 'Enter') {
+      this.setState({ isCalendarOpenedViaKeyboard: true })
```

Comment:
If the picker was launched using the `Enter` key once, `this.state.isCalendarOpenedViaKeyboard` always remains true even when you click to open it. As a result, the preselected day will get the dash border even when you open it via clicking on the calendar button.

Reply:
Fixed cbabd6c